File: 프로그래머스/1/42840. 모의고사/모의고사.py
def grading(case, answers):
    choice = []
    ## 수포자별 정답지
    if case == 1: choice = [1,2,3,4,5]
    if case == 2: choice = [2,1,2,3,2,4,2,5]
    if case == 3: choice = [3,3,1,1,2,2,4,4,5,5]
    
    ## 채점 후 점수 반환
    answers = [1 if answers[idx]==choice[idx%len(choice)] else 0 for idx in range(len(answers))]
    return sum(answers)

def solution(answers):
    case_1, case_2, case_3 = grading(1, answers), grading(2, answers), grading(3, answers)
    
    ## 최고 득점이 같다면, 번호 순으로 정렬
    max_score = max([case_1, case_2, case_3])
    answer = list(filter(lambda x: x[1]==max_score, [(1,case_1),(2,case_2),(3,case_3)]))
    return list(map(lambda x:x[0], answer))
    

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>

using namespace std;

int person1[5] = {1, 2, 3, 4, 5}, cnt1 = 0;
int person2[8] = {2, 1, 2, 3, 2, 4, 2, 5}, cnt2 = 0;
int person3[10] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}, cnt3 = 0;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    for(int i=0; i<answers.size(); i++){
        if(person1[i%5] == answers[i]) cnt1++;
        if(person2[i%8] == answers[i]) cnt2++;
        if(person3[i%10] == answers[i]) cnt3++;
    }
    
    if((cnt1 == cnt2) && (cnt2 == cnt3)) return {1,2,3};
    if((cnt1 == cnt2) && (cnt2 > cnt3)) return {1,2};
    if((cnt1 == cnt3) && (cnt3 > cnt2)) return {1,3};
    if((cnt2 == cnt3) && (cnt2 > cnt1)) return {2,3};
    if((cnt1 > cnt2) && (cnt1 > cnt3)) return {1};
    if((cnt2 > cnt1) && (cnt2 > cnt3)) return {2};
    if((cnt3 > cnt2) && (cnt3 > cnt1)) return {3};
}